iron-dropdown.html

```diff
@@ -340,6 +340,49 @@
           } else {
             Polymer.IronOverlayBehaviorImpl._applyFocus.apply(this, arguments);
           }
+        },
+
+        /**
+         * An element related to the dropdown (the triggering element, or an
+         * element within the dropdown) has lost focus.
+         */
+        _activeElementBlurred: function(event) {
+          if (event.target._blurListener) {
+            // Stop listening to blur events on the target.
+            event.target.removeEventListener('blur', event.target._blurListener);
+            event.target._blurListener = null;
+          }
+          this._dropdownFocusChanged(false);
+        },
+
+        /**
+         * Return true if the focus is anywhere within the dropdown.
+         */
+        _focusWithinDropdown: function() {
+          if (this.shadowRoot) {
+            return this.shadowRoot.activeElement !== null;
+          } else {
+            var activeElement = this._manager.deepActiveElement;
+            return Polymer.dom(this).deepContains(activeElement);
+          }
+        },
+
+        /**
+         * Overridden from `IronOverlayBehavior`.
+         */
+        _onCaptureFocus: function(event) {
+          if (!this._focusWithinDropdown()) {
+            // The focus has left the dropdown, e.g., in response to the user
+            // tabbing away.
+
+            // Avoid restoring focus after is closed.
+            this._restoreFocusNode = null;
```

Comment:
Now that I think of it, over here we are disabling focus restoring for a lot of other events that cause focus change, e.g. click outside, press Escape key to close 😞 
Do we want to disable it also for those events? That would mean that any event that causes a canceling of the overlay should NOT restore focus (and that should be fixed in `iron-overlay-behavior`).
If we want to only disable it only for focus, I'd suggest to override in here the `cancel` method to something like this:
```javascript
// Overridden from `IronOverlayBehavior`.
cancel: function(event) {
  Polymer.IronOverlayBehaviorImpl.cancel.apply(this, arguments);
  // If it is a focus event and the overlay has been closed
  if (event.type === 'focus' && !this.opened) {
    // Avoid restoring focus after is closed.
    this._restoreFocusNode = null;
  }
}
```

WDYT? cc/ @cdata